node-export-server/server.js

```diff
@@ -129,6 +133,78 @@ app.get('/export/:datasource/:schema/:format', function(req, res) {
 });
 
 exports.writeExportFile = writeExportFile
+
+// ensures poly string in query params
+//  1. follows structure lon_0,lat_0;...lon_n,lat_n;lon_0,lat_0
+//  2. first and last match, meaning its a polygon
+//  3. does not self intersect as deterimed by shamos-hoey alg
+exports.validatePoly = function(poly) {
+    if (!poly) return null
+    var match = poly.split(';');
+
+    function validCoordinates(coordinates) {
+        return coordinates.findIndex(function(c) { // if the polygon coordinates are valid
+            return (c[0] >= -180 && c[0] <= 180)
+                && (c[1] >= -90 && c[1] <= 90);
+        }) !== -1
+    }
+
+    function matchingFirstLast(coordinates) {
+        return coordinates[0][0] === coordinates[coordinates.length - 1][0] &&
+               coordinates[0][1] === coordinates[coordinates.length - 1][1]
+    }
+
+    if (match.length > 1) {
+        var coordinates = []
+        for (m of match) {
+            // gather capture groups to add to coordinates array
+            // if we ever don't get a capture, exit early.
+            // this is a way to make sure nothing that isn't a cooridnate can get injected into the hoot command
+            var captures = /(-?\d+\.?\d*),(-?\d+\.?\d*)/g.exec(m);
+            if (captures) {
+                coordinates.push(captures.splice(1).map(parseFloat))
+            } else {
+                return null
+             }
+        }
+
+        if (
+            validCoordinates(coordinates) && matchingFirstLast(coordinates) &&
```

Comment:
are there other things to validate against?

we make sure its an array of coordinates in between -+180/90 deg, that the first in last coordinates match, and there are no self intersections.

perhaps the order coordinates are wound matters?

Reply:
I don't think clockwise v. counterclockwise winding direction matters. I just tried same polygon for the gif below with a flipped direction and it returned the same output.